packages/babel-plugin-jest-hoist/src/index.ts

```diff
@@ -152,38 +165,140 @@ FUNCTIONS.deepUnmock = args => args.length === 1 && args[0].isStringLiteral();
 FUNCTIONS.disableAutomock = FUNCTIONS.enableAutomock = args =>
   args.length === 0;
 
-export default (): {visitor: Visitor} => {
-  const shouldHoistExpression = (expr: NodePath): boolean => {
-    if (!expr.isCallExpression()) {
-      return false;
-    }
+const createJestObjectGetter = statement`
+function GETTER_NAME() {
+  const { JEST_GLOBALS_MODULE_JEST_EXPORT_NAME } = require("JEST_GLOBALS_MODULE_NAME");
+  GETTER_NAME = () => JEST_GLOBALS_MODULE_JEST_EXPORT_NAME;
+  return JEST_GLOBALS_MODULE_JEST_EXPORT_NAME;
```

Comment:
@jeysal Even if this was my original suggestion, I [recently learned](https://github.com/airbnb/babel-plugin-dynamic-import-node/pull/86#discussion_r411712880) that, since `require` is cached, we might not need to manually cache this function. This might not apply depending on `jest`'s implementation of `require`, but if you have some perf tests it might be worth checking it.

Reply:
Hmm, even then, isn't it probably slightly better to avoid the `require` and property access altogether and store the `jest` var in the closure of the new function instead?

Reply:
`require` is cached, but that cache can be cleared (`delete require.cache` in node and `jest.resetModules()` in jest). I like avoiding the extra function call if it's not needed